src/objects/uiObject.hpp

```diff
@@ -1,16 +1,21 @@
 #pragma once
 
 #include "manyakSDL.hpp"
+#include "texture.hpp"
+#include "vec2.hpp"
 
 class UIObject {
 
 public:
 	// TODO Just by logic, UIObjects will have children. This object should have children and should render them as well.
-	void render();
+	virtual void render();
+	virtual void update();
 
 	virtual ~UIObject();
 protected:
 
-	SDL_Texture* mTexture;
+	Texture* mTexture = nullptr;
+	Vec2 mPosition;
+	SDL_Texture* mSDL_Texture = nullptr;
 	SDL_Rect mRenderBox;
```

Comment:
RenderBox holds both the position and the size. Either get rid of the renderbox here or get rid of position and size.

Reply:
Resolved!